### DISCUSSION TITLE
[IOS] is it possible to retrieve a history of consumables / renewable subscriptions transactions?

### DISCUSSION THREAD

**Discussion by vesper8** [imdhemy/laravel-in-app-purchases#93] [Ideas]
I'm still in Sandbox mode right now, and I understand how important it is to keep a log of every transaction. And I have built these mechanisms. But I also have now truncated these tables since I'm still developing and testing and I find myself in a position where I know this particular user has purchases multiple consumables, and currently has one of the renewable-subscriptions set to owned. But I do not have any records of these on my back-end.. because I purposefully truncated that data.

I'm just curious, without a log of transaction receipts or transaction ids or anything basically.. is it possible to retrieve a history of transactions for a given user?

I would like to know for example what is the expiry date for the currently owned renewable subscription.

And I would like to see how many consumables this user has purchased.

If I don't have the log.. is this data available anywhere else?

Is it also possible to have a little more visibility on transactions performed inside the sandbox environment? I can't seem to find that anywhere on appstoreconnect

Many thanks

**Comment by imdhemy** [comment]
@vesper8 Thanks for starting this discussion, AFAIK, both providers (Google and Apple) keeps a history for all transactions and you can access them from the CP. Even though this is not enough for the back-end and we still can not find which transaction belongs to which user on our application. But still enough for customer support.

For Apple, you can get the history of all transactions from their API, but this still requires a purchase receipt to send the request.

IMHO, keeping a log for every transaction is the best solution we can do. It would be great if Apple and Google allowed us to add some extra data to the purchase event, something like the user ID at our end for instance, besides an API to retrieve the log. It would be a lifesaver for us.
